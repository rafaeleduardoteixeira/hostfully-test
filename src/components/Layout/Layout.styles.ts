import styled from 'styled-components';
import { media } from '../../themes';

export const Container = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 25px 0;
  width: 100%;
`;

export const Logo = styled.img`
  width: 180px;
  height: auto;
  cursor: pointer;

  @media ${media.tablet} {
    width: 220px;
  }

  @media ${media.mobile} {
    width: 150px;
  }
`;
