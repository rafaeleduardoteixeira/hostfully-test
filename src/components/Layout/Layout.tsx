import { Outlet, ScrollRestoration } from 'react-router-dom';
import { HeaderNav } from '../HeaderNav/HeaderNav';
import { Container, Logo } from './Layout.styles';

export const Layout = (): JSX.Element => {
  return (
    <>
      <Container>
        <a href="/">
          <Logo src="/icons/logo.svg" alt="logo" />
        </a>
        <HeaderNav />
      </Container>
      <Outlet />
      <ScrollRestoration />
    </>
  );
};
