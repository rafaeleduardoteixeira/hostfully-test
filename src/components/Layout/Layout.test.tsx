import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Layout } from './Layout';

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => jest.fn(),
  ScrollRestoration: () => jest.fn(),
}));

describe('Layout', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<Layout />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Render', () => {
    const { getByAltText } = render(<Layout />);
    expect(getByAltText('logo')).toBeInTheDocument();
  });
  test('Logo click', () => {
    const { getByAltText } = render(<Layout />);
    const logo = getByAltText('logo');
    logo.click();
  });
});
