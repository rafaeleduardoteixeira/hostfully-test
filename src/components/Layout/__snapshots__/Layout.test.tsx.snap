// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshot test 1`] = `
<DocumentFragment>
  <header
    class="sc-iKTcqh evAMOK"
  >
    <img
      alt="logo"
      class="sc-gnpbhQ dFwobQ"
      src="/icons/logo.svg"
    />
    <div
      class="sc-braxZu cnSkOo"
      data-testid="HeaderNavMenuClick"
    >
      <img
        class="sc-gJhJTp cboMkP"
        src="/icons/header-nav.svg"
      />
      <img
        class="sc-elDIKY RUxgM"
        src="https://www.rafaeleteixeira.com.br/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fprofile.a4caa42b.jpg&w=3840&q=75"
      />
    </div>
  </header>
</DocumentFragment>
`;
